src/uptime.rs

```diff
@@ -0,0 +1,115 @@
+use std::fs::read_to_string;
+
+use serde::{Deserialize, Serialize};
+
+use crate::Config;
+use crate::Module;
+
+const UPTIME_PATH: &str = "/proc/uptime";
+
+#[derive(Serialize, Deserialize)]
+#[serde(default)]
+pub struct Uptime {
+    header: String,
+
+    #[serde(skip)]
+    /// Uptime in seconds
+    uptime: u32,
+
+    /// When true, uptime will be outputted like: 2d 10h 3m
+    short_output: bool,
+}
+
+/// Returns system uptime in seconds
+fn get_uptime() -> u32 {
+    read_to_string(UPTIME_PATH)
+        .expect("Failed to read uptime file")
+        .split_once(' ')
+        .unwrap()
+        .0
+        .parse::<f32>()
+        .expect("Failed to parse uptime") as u32
+}
+
+/// Converts seconds into days, hours, minutes
+fn days_hours_mins(secs: u32) -> (u32, u32, u32) {
+    let days = secs / 60 / 60 / 24;
+    let hours = secs / 60 / 60 % 24;
+    let minutes = secs / 60 % 60;
```

Comment:
i think we can have some consts for these:
(is `%` really needed since `u32` does not have decimals?)
```
const SECS_PER_MINUTE: u32 = 60;
const SECS_PER_HOUR: u32 = 3600;
const SECS_PER_DAY: u32 = 86400;
```

Reply:
> i think we can have some consts for these

Yeah, maybe that would be cleaner. I just figured since these numbers are not really bound to change anytime soon 😄

> (is % really needed since u32 does not have decimals?)

Yes it is actually necessary, since we need the remainder from the days division to calculate the hours and the remainder from the hours to calculate the minutes.

Reply:
ahh yes ofc :facepalm: